frontend/src/metabase/actions/containers/ActionCreator/tests/ActionCreator-Sharing.unit.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+
+import { screen, waitFor } from "__support__/ui";
+
+import {
+  createMockImplicitQueryAction,
+  createMockQueryAction,
+} from "metabase-types/api/mocks";
+
+import { setup as baseSetup, SITE_URL, SetupOpts } from "./common";
+
+async function setup({
+  action = createMockImplicitQueryAction(),
+  ...opts
+}: SetupOpts = {}) {
+  await baseSetup({ action, ...opts });
+  return { action };
+}
+
+describe("ActionCreator > Sharing", () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  describe.each([
+    ["query", createMockQueryAction],
+    ["implicit", createMockImplicitQueryAction],
+  ])(`%s actions`, (_, getAction) => {
+    describe("admin users and has public sharing enabled", () => {
+      const mockUuid = "mock-uuid";
+      const privateAction = getAction();
+      const publicAction = getAction({ public_uuid: mockUuid });
+
+      it("should show action settings button", async () => {
+        await setup({
+          action: privateAction,
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        expect(
+          screen.getByRole("button", { name: "Action settings" }),
+        ).toBeInTheDocument();
+      });
+
+      it("should be able to enable action public sharing", async () => {
+        await setup({
+          action: privateAction,
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        screen.getByRole("button", { name: "Action settings" }).click();
+
+        expect(screen.getByText("Action settings")).toBeInTheDocument();
+        const makePublicToggle = screen.getByRole("switch", {
```

Comment:
One thing worth noting is that, I extracted this element into a variable so we don't have to do the look-up over and over again. This caused the unit test to flake in the past somehow, [since `*ByRole` could be slow](https://github.com/testing-library/dom-testing-library/issues/552).